package-res/resources/web/prompting/PromptPanel.js

```diff
@@ -1209,6 +1212,14 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
                     this.dashboard.updateComponent(component);
                   }.bind(this));
                 }
+
+                if ((param.attributes["post-processor-formula"] || param.attributes["default-value-formula"]) && param.getSelectedValuesValue().length > 0) {
```

Comment:
This looks as if there is something else going on. The result of the post-processing is a new selected value and that value should have been present in the values array and thus be set via the "this._initializeParameterValue(null, param)" call earlier on. 

If the code above does not catch the case where the parameter-set (the elements available to be selected) remains the same, but the selection set (the values of the parameter-set that are actually selected) changes, then fix it there. I would also assume that this kind of bug would be independent of whether the parameter has a post-processing or default-value formula. 

You should be able to validate that by writing a unit-test that triggers that condition.